### PR TITLE
Remove support for specializing ToString outside the standard library

### PR DESCRIPTION
**Pull request by bjorn3**
This is the only trait specializable outside of the standard library. Before stabilizing specialization we will probably want to remove support for this. It was originally made specializable to allow a more efficient ToString in libproc_macro back when this way the only way to get any data out of a TokenStream. We now support getting individual tokens, so proc macros no longer need to call it as often.

### CODE REVIEW

File: library/proc_macro/src/lib.rs [hunk 1]
```diff
@@ -19,9 +19,6 @@
 )]
 #![doc(rust_logo)]
 #![feature(rustdoc_internals)]
-// This library is copied into rust-analyzer to allow loading rustc compiled proc macros.
-// Please avoid unstable features where possible to minimize the amount of changes necessary
-// to make it compile with rust-analyzer on stable.
```
Comment (bjorn3) [inline library/proc_macro/src/lib.rs:24]: This hasn't been the case anymore for years now.

**Comment by lqd** [comment]
@rust-timer build f557de4c3dacd27cfe1deb8a6286a175e03ea4b8

**Comment by petrochenkov** [comment]
Perf is ok, but the CI is still failing.
@rustbot author

**Pull request merged**